controllers/provisioning_controller.go

```diff
@@ -655,7 +662,6 @@ func (r *ProvisioningReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&appsv1.DaemonSet{}).
 		Owns(&osconfigv1.ClusterOperator{}).
 		Owns(&osconfigv1.Proxy{}).
-		Owns(&machinev1beta1.Machine{}).
```

Comment:
I agree that this was definitely having no useful effect in terms of triggering reconciles, since the Provisioning CR (thankfully) never owns any Machines.

The correct thing here I think would be to conditionally add `Watches(&machinev1beta1.Machine{}, handler.EnqueueRequestsFromMapFunc(watchOCPConfigPullSecret)` (note the map function could use a better name. Also the next line is [wrong as well](https://github.com/openshift/cluster-baremetal-operator/pull/351/files#r1628776168) I think.)

If removing this also has the effect of no longer caching Machines (I can't remember if this is the case, but it seems likely) then we won't actually ever be able to find the Machine when we look it up (even on hosts where the CRD exists), assuming we use the cached client.

Reply:
In the log I see:

```
2024-06-05T15:23:12.873346427Z W0605 15:23:12.873314       1 provisioning_controller.go:521] Machine cluster13-wzd5l-master-1 has no annotation linking it to BareMetalHost
2024-06-05T15:23:12.873392927Z W0605 15:23:12.873383       1 provisioning_controller.go:521] Machine cluster13-wzd5l-master-2 has no annotation linking it to BareMetalHost
2024-06-05T15:23:12.873420905Z W0605 15:23:12.873412       1 provisioning_controller.go:521] Machine cluster13-wzd5l-master-0 has no annotation linking it to BareMetalHost
2024-06-05T15:23:12.974889888Z I0605 15:23:12.974839       1 provisioning_controller.go:601] "Updating provisioning MAC addresses" CurrentMacAddresses=null NewMacAddresses=["F8:F2:1E:B2:DB:51","F8:F2:1E:B2:E7:81","F8:F2:1E:B2:E9:D1"]
```

so I guess this is not preventing us reading the Machines at least.

Reply:
Just to make sure, we want to have the Machine in the client cache (if the Machine CR is registered) we don't want to trigger reconcile, right?

Reply:
Provided you are using the standard client that controller runtime provides, the first time an object type is listed or fetched, the controller will start to cache objects of that kind across the cluster.

There are options you can use to configure this behaviour, such as excluding certain types from being cached or restricting caching to all namespaces, best to check you aren't using any of those presently, but, the `Owns` here won't affect whether your machine objects are cached or not